src/main/java/ticketbooking/web/controller/OrderController.java

```diff
@@ -0,0 +1,208 @@
+package ticketbooking.web.controller;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import ticketbooking.model.Order;
+import ticketbooking.model.Screening;
+import ticketbooking.model.Seat;
+import ticketbooking.model.Ticket;
+import ticketbooking.service.ClientService;
+import ticketbooking.service.AppUserService;
+import ticketbooking.service.OrderService;
+import ticketbooking.service.ScreeningService;
+import ticketbooking.service.SeatService;
+import ticketbooking.service.TicketService;
+import ticketbooking.struct.TicketType;
+import ticketbooking.web.CreateOrderRequest;
+import ticketbooking.web.CreateTicketRequest;
+import ticketbooking.web.OrderRequest;
+import ticketbooking.web.OrderResponse;
+import ticketbooking.web.exception.ScreeningTimeException;
+import ticketbooking.web.exception.SeatException;
+import ticketbooking.web.exception.SeatScreeningException;
+import ticketbooking.web.exception.SeatDuplicateException;
+import ticketbooking.web.exception.OrderException;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrderController.class);
+
+    private OrderService orderService;
+
+    private AppUserService appUserService;
+
+    private ClientService clientService;
+
+    private ScreeningService screeningService;
+
+    private SeatService seatService;
+
+    private TicketService ticketService;
+
+    public OrderController(OrderService orderService, AppUserService appUserService, ClientService clientService, ScreeningService screeningService, SeatService seatService, TicketService ticketService) {
```

Comment:
nie za dużo zależności?

Reply:
Jak są potrzebne to jest dużo. Jak masz pomysł jak to zrobić lepiej to z chęcią posłucham.